single_kernel_mongo/managers/backups.py

```diff
@@ -442,46 +456,57 @@ def get_backup_error_status(self, backup_id: str) -> str:
 
         return ""
 
-    def process_pbm_error(self, pbm_status: str) -> str:
+    def process_pbm_error(self, pbm_status: str) -> StatusBase | None:
         """Look up PBM status for errors."""
         error_message: str
-        message = ""
+
         try:
             pbm_as_dict = json.loads(pbm_status)
             error_message = self.retrieve_error_message(pbm_as_dict)
         except json.JSONDecodeError:
             error_message = pbm_status
 
         if StatusCodeError.FORBIDDEN in error_message:
-            message = "s3 credentials are incorrect."
-        elif StatusCodeError.NOTFOUND in error_message:
-            message = "s3 configurations are incompatible."
-        elif StatusCodeError.MOVED_PERMANENTLY in error_message:
-            message = "s3 configurations are incompatible."
-        return message
+            return BackupStatuses.PBM_INCORRECT_CREDS.value
+        if StatusCodeError.NOTFOUND in error_message:
+            return BackupStatuses.PBM_INCOMPATIBLE_CONF.value
+        if StatusCodeError.MOVED_PERMANENTLY in error_message:
+            return BackupStatuses.PBM_INCOMPATIBLE_CONF.value
+
+        if error_message:
+            logger.info("PBM error: %s", error_message)
+            return BackupStatuses.UNKNOWN_PBM_ERROR.value
+
+        return None
 
     def process_pbm_status(self, pbm_status: str) -> StatusBase:
         """Processes the pbm status if there's no error."""
         pbm_as_dict: dict[str, dict] = json.loads(pbm_status)
         current_op = pbm_as_dict.get("running", {})
         match current_op:
             case {"type": "backup", "name": backup_id}:
-                return MaintenanceStatus(f"backup started/running, backup id: '{backup_id}'")
+                return BackupStatuses.backup_running(backup_id)
             case {"type": "restore", "name": backup_id}:
-                return MaintenanceStatus(f"restore started/running, backup id: '{backup_id}'")
+                return BackupStatuses.restore_running(backup_id)
             case {"type": "resync"}:
-                return WaitingStatus("waiting to sync s3 configurations.")
+                return BackupStatuses.PBM_WAITING_TO_SYNC.value
             case _:
-                return ActiveStatus("")
+                return BackupStatuses.ACTIVE_IDLE.value
 
     def assert_can_restore(self, backup_id: str, remapping_pattern: str) -> None:
         """Does the status allow to restore.
 
+        Note: we permit this logic based on status since we aren't checking
+        `self.charm.unit.status`, instead `get_status` directly computes the status of pbm.
+
         Returns:
             check: boolean telling if the status allows to restore.
             reason: The reason if it is not possible to restore yet.
         """
-        pbm_status = self.get_status()
+        pbm_statuses = self.get_statuses()
+        pbm_status = next(iter(pbm_statuses), None)
+
+        # todo future work - check status of pbm directly
```

Comment:
don't we already have `_get_backup_restore_operation_result` to get the information from?